### PR TITLE
*: optimize analyze cluster index table

### LINKED ISSUE #20243: Optimize analyze for clustered index
## Development Task

Currently analyze clustered index request two full table scan, one is for analyzing the primary key index, another one is for analyzing the columns in the row. This is due to easier to implement.

We need to avoid the extra full table scan by analyzing the primary key index and row columns at the same time.

### PR DESCRIPTION
**Pull request by lzmhhh123**
### What problem does this PR solve?

Issue Number: close #20243

Currently, we analyze clustered index and columns by scan the same key-value pairs twice to do different analyze tasks. After this PR, coprocessor only need to scan them once.

### Related changes

- https://github.com/pingcap/tipb/pull/197
- https://github.com/tikv/tikv/pull/9175

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- none.

### CODE REVIEW

File: store/mockstore/unistore/cophandler/analyze.go [hunk 1]
```diff
@@ -350,3 +352,172 @@ func (e *analyzeColumnsExec) NewChunk() *chunk.Chunk {
 func (e *analyzeColumnsExec) Close() error {
 	return nil
 }
+
+func handleAnalyzeMixedReq(dbReader *dbreader.DBReader, rans []kv.KeyRange, analyzeReq *tipb.AnalyzeReq, startTS uint64) (*coprocessor.Response, error) {
```
Comment (coocood) [inline store/mockstore/unistore/cophandler/analyze.go:356]: Can we reuse any existing analyze code for mixed analyze logic?

Comment (lzmhhh123) [reply store/mockstore/unistore/cophandler/analyze.go:363]: I have extracted a function `buildBaseAnalyzeColumnsExec` to reuse some codes.

File: store/mockstore/unistore/cophandler/analyze.go [hunk 2]
```diff
@@ -350,3 +359,123 @@ func (e *analyzeColumnsExec) NewChunk() *chunk.Chunk {
...
+		}
+	}
+	rowData := safeCopy(e.rowBuf)
+	err = e.statsBuilder.Iterate(types.NewBytesDatum(rowData))
+	if err != nil {
+		return err
+	}
+
+	// columns
+	handle, err := tablecodec.DecodeRowKey(key)
```
Comment (coocood) [inline store/mockstore/unistore/cophandler/analyze.go:434]: Can we just call `e.analyzeColumnsExec.Process`?

Comment (lzmhhh123) [reply store/mockstore/unistore/cophandler/analyze.go:434]: Of course, we can.

**Comment by coocood** [comment]
LGTM

**Comment by hi-rustin** [comment]
@ti-chi-bot update this pr

**Pull request merged**